File: persistence/src/main/java/io/sparkled/persistence/playlist/impl/query/GetPlaylistByIdQuery.java
package io.sparkled.persistence.playlist.impl.query;

import io.sparkled.model.entity.Playlist;
import io.sparkled.persistence.PersistenceQuery;
import io.sparkled.persistence.QueryFactory;

import java.util.Optional;

public class GetPlaylistByIdQuery implements PersistenceQuery<Optional<Playlist>> {

    private final int playlistId;

    public GetPlaylistByIdQuery(int playlistId) {
        this.playlistId = playlistId;
    }

    @Override
    public Optional<Playlist> perform(QueryFactory queryFactory) {
        final Playlist playlist = queryFactory
                .selectFrom(qPlaylist)
                .where(qPlaylist.id.eq(playlistId))
                .fetchFirst();

        return Optional.ofNullable(playlist);
    }
}

File: persistence/src/main/kotlin/io/sparkled/persistence/playlist/impl/query/GetPlaylistByIdQuery.kt
package io.sparkled.persistence.playlist.impl.query

import io.sparkled.model.entity.Playlist
import io.sparkled.model.entity.QPlaylist.Companion.playlist
import io.sparkled.persistence.PersistenceQuery
import io.sparkled.persistence.QueryFactory

class GetPlaylistByIdQuery(private val playlistId: Int) : PersistenceQuery<Playlist?> {

    override fun perform(queryFactory: QueryFactory): Playlist? {
        return queryFactory
            .selectFrom(playlist)
            .where(playlist.id.eq(playlistId))
            .fetchFirst()
    }
}
